### PR TITLE
Add reporting of MXNet version

### PR DESCRIPTION
**Pull request by jens-mueller-sociomantic**
This change provides a function to obtain the MXNet version of the
linked library. We also output this version as a unittest. The
diagnostic can be useful when checking which MXNet version is actually
used.

### CODE REVIEW

File: src/mxnet/API.d [hunk 1]
```diff
@@ -157,3 +159,150 @@ public void notifyShutdown ()
...
+        Patch version
+
+    ***************************************************************************/
+
+    ubyte patch;
+
+
+    /***************************************************************************
+
+        Formats this version
```
Comment (daniel-zullo) [inline src/mxnet/API.d]: You might want to change it to `Formats the MXNet version`

File: src/mxnet/API.d [hunk 2]
```diff
@@ -157,3 +159,150 @@ public void notifyShutdown ()
...
+
+public struct MXNetVersion
+{
+    /***************************************************************************
+
+        Major version
+
+    ***************************************************************************/
+
+    uint major;
```
Comment (daniel-zullo) [inline src/mxnet/API.d:194]: would it be possible to use `///` to document member variables or single documentation lines?

**Review by daniel-zullo — COMMENTED** [review]
I only have a cosmetic suggestion and a question and LGTM

File: src/mxnet/API.d [hunk 3]
```diff
@@ -157,3 +159,150 @@ public void notifyShutdown ()
...
+        major = specifies the major version
+        minor = specifies the minor version
+        patch = specifies the patch version
+
+    Returns:
+        the specified MXNet version
+
+*******************************************************************************/
+
+private MXNetVersion mxnetVersion (uint major, ubyte minor, ubyte patch)
```
Comment (stefan-koch-sociomantic) [inline src/mxnet/API.d]: maybe you could get rid of this helper?

Comment (jens-mueller-sociomantic) [reply src/mxnet/API.d]: Why? I'm using it in the tests.

File: src/mxnet/API.d (hunk 2 above)
Comment (jens-mueller-sociomantic) [reply src/mxnet/API.d:194]: I could.

File: src/mxnet/API.d (hunk 3 above)
Comment (stefan-koch-sociomantic) [reply src/mxnet/API.d]: it does not look like you really need it.
You can do the modding and division in struct literal as well, no?

Comment (jens-mueller-sociomantic) [reply src/mxnet/API.d]: I don't see it.

Comment (jens-mueller-sociomantic) [reply src/mxnet/API.d]: Clarified in real life. The suggestion is to use the implicit constructor.

File: src/mxnet/API.d [hunk 4]
```diff
@@ -157,3 +159,108 @@ public void notifyShutdown ()
...
+{
+    // major * 10_000 + minor * 100 + patch
+    // see `MXNET_VERSION` is defined in `include/mxnet/base.h`
+    MXNetVersion version_ =
+    {
+        major : mxnet_version / 10_000,
+        minor : mxnet_version % 10_000 / 100,
+        patch : mxnet_version % 100,
+    };
+    return version_;
```
Comment (jens-mueller-sociomantic) [inline src/mxnet/API.d:244]: I could also do

```d
return MXNetVersion(mxnet_version / 10_000,
                    mxnet_version % 10_000 / 100,
                    mxnet_version % 100);
```

but with the `{ ... }` it looks more explicit though easier to break when we add a field.

Comment (daniel-zullo) [reply src/mxnet/API.d:244]: The current version is easier to read and it is unlikely a new field will be added to the version

**Review by daniel-zullo — DISMISSED** [review]
LGTM

File: src/mxnet/API.d (hunk 4 above)
Comment (jens-mueller-sociomantic) [reply src/mxnet/API.d:244]: I'll agree.

**Comment by jens-mueller-sociomantic** [comment]
Add missing release notes.

**Review by UplinkCoder — APPROVED** [review]
LGTM

**Review by daniel-zullo — APPROVED** [review]

**Pull request merged**